### PR TITLE
NEW entry points / new main script / simple docs

### LINKED ISSUE #336: The new user interface
There needs to be a rework of the beast user interface. We can use an entry point but it needs to point to a new python script that has arguments to run other scripts.

To run the beast we should be able to initiate commands outside of python using:
    `beast `
    `beast production`
    `beast plot_figs`
     etc.

Adding this new script to the entry point requires changing entry_points in setup.py, i.e.:

`entry_points = {'console_scripts': ['beast = beast.new_script.main:']}`

### PR DESCRIPTION
**Pull request by s-goldman**
Closes #336.

### CODE REVIEW

**Comment by karllark** [comment]
Please remove the unused imports identified by LGTM when you get a chance.

**Comment by karllark** [comment]
Is this PR ready to merge?  If so, please remove the WIP tag in the PR title.

File: main.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+def main():
+    """Summary gets arguments for each of the scripts listed in 'scripts' 
+    and parses them for the function given in the input
+    """
+    all_funcs = []
+    scripts = [plot_cmd, plot_filters, plot_indiv_fit]
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--global-setting", action="store_true", help="some global thingie"
```
Comment (karllark) [inline main.py]: A better help string is needed.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            # make a dictionary containing the correct inputs to the function,
+            # extracted from the parsed arguments
+            funcargs = {}
+            for parname in inspect.signature(func).parameters:
+                funcargs[parname] = getattr(args, parname.replace("-", "_"))
+            func(**funcargs)
+            break  # drop out immediately, which skips the "else" below
+    else:
+        if False:
+            raise AssertionError("Invalid subparser! This should be impossible...")
```
Comment (karllark) [inline main.py]: Will never be reached as "if False" never is true.  Do you even need the "if False" line?

**Comment by s-goldman** [comment]
The "unused" functions are being used, codacy is a liar. It's just that they are called as items in a list and therefore not explicitly mentioned in the text.

**Comment by s-goldman** [comment]
Also not sure why the setup.py script is saying there is an entrypoint syntax error?

**Pull request closed without merging**